adhocracy4/comments/serializers.py

```diff
@@ -10,16 +10,24 @@ class CommentSerializer(serializers.ModelSerializer):
     user_name = serializers.SerializerMethodField()
     user_profile_url = serializers.SerializerMethodField()
     is_deleted = serializers.SerializerMethodField()
-    ratings = serializers.SerializerMethodField()
+    ratings = serializers.ModelSerializer(many=True)
     is_moderator = serializers.SerializerMethodField()
+    comment_categories = serializers.ModelSerializer(many=True)
+    child_comments = serializers.ModelSerializer(many=True)
 
     class Meta:
         model = Comment
         read_only_fields = ('modified', 'created', 'id',
                             'user_name', 'ratings', 'content_type',
                             'object_pk', 'last_discussed',
                             'is_moderator_marked')
-        exclude = ('creator',)
+        fields = ('id', 'user_name', 'user_pk', 'user_profile_url',
```

Comment:
@Rineee @fuzzylogic2000  as we discussed in stand-up, I want to replace the exclude in the comment serializer with the defined fields. Checking the model for field names didn't work for me, so I checked what the api receives without restrictions by setting ...fields = '__all__'.... This returned to me the list ...{
            "id": 9,
            "user_name": "sabina-follower",
            "user_pk": "5",
            "user_profile_url": "/profile/sabina-follower/",
            "user_image": "/static/images/avatar-01.svg",
            "is_deleted": false,
            "ratings": {
                "positive_ratings": 0,
                "negative_ratings": 0,
                "current_user_rating_value": null,
                "current_user_rating_id": null
            },
            "is_moderator": false,
            "child_comments": [],
            "created": "2021-11-03T16:26:36.983486+01:00",
            "modified": null,
            "object_pk": 1,
            "comment": "afadfdafdafa",
            "is_removed": false,
            "is_censored": false,
            "is_blocked": null,
            "comment_categories": {},
            "last_discussed": null,
            "is_moderator_marked": false,
            "creator": 5,
            "content_type": 112
        },... BUT now tests fail because e.g. Field name `user_pk` is not valid for model `Comment`. This makes sense because it's not in the model. Where does it come from?

Reply:
@sabinammm The user_pk, user_name etc come from UserGeneratedContentModel, the model that Comment class inherits from.

I dont get that error if I add the fields you added.. could you check again?
I get another error though for the child_comments, because they are added in the ThreadSerializer (this is the one we are actually using to get the child comments) below, but not explicitly in the fields list. You can solve this by adding 
`
    class Meta(CommentSerializer.Meta):
        model = Comment
        fields = CommentSerializer.Meta.fields + ('child_comments',)
`
to both the ThreadSerializer and ThreadListSerializer, so the child_comments are added to fields list there.

Reply:
> @sabinammm The user_pk, user_name etc come from UserGeneratedContentModel, the model that Comment class inherits from.
> 
> I dont get that error if I add the fields you added.. could you check again? 
@Rineee I mean this error: https://github.com/liqd/adhocracy4/runs/4116676609?check_suite_focus=true#step:10:209

I get another error though for the child_comments, because they are added in the ThreadSerializer (this is the one we are actually using to get the child comments) below, but not explicitly in the fields list. You can solve this by adding `
> 
> class Meta(CommentSerializer.Meta):
> 
> ```
> model = Comment
> 
> fields = CommentSerializer.Meta.fields + ('child_comments',)
> ```
> 
> ` to both the ThreadSerializer and ThreadListSerializer, so the child_comments are added to fields list there.

@Rineee yes! will do this, thanks!

Reply:
yes!